pints/toy/_hes1_michaelis_menten.py

```diff
@@ -50,14 +49,75 @@ class Hes1Model(pints.ForwardModel, ToyModel):
            https://doi.org/10.1038/ncomms1496
     """
     def __init__(self, y0=None, implicit_parameters=None):
-        if y0 is None:
-            self.set_initial_conditions(2)
-        else:
-            self.set_initial_conditions(y0)
         if implicit_parameters is None:
             self.set_implicit_parameters([5., 3., 0.03])
         else:
             self.set_implicit_parameters(implicit_parameters)
+        if y0 is None:
+            self.set_initial_conditions(2)
+        else:
+            self.set_initial_conditions(y0)
+
+    def _dfdp(self, state, time, parameters):
+        """ See :meth:`pints.ToyModel.jacobian()`. """
+        m, p1, p2 = state
+        P0, v, k1, h = parameters
+        p2_over_p0 = p2 / P0
+        p2_over_p0_h = p2_over_p0**h
+        one_plus_p2_expression_sq = (1 + p2_over_p0_h)**2
+        ret = np.empty((self.n_states(), self.n_parameters()))
+        ret[0, 0] = h * p2 * p2_over_p0**(h - 1) / (
+            P0**2 * one_plus_p2_expression_sq)
+        ret[0, 1] = 0
+        ret[0, 2] = 0
+        ret[0, 3] = - (p2_over_p0_h * np.log(p2_over_p0)) / (
+            one_plus_p2_expression_sq
+        )
+        ret[1, 0] = 0
+        ret[1, 1] = m
+        ret[1, 2] = -p1
+        ret[1, 3] = 0
+        ret[2, 0] = 0
+        ret[2, 1] = 0
+        ret[2, 2] = p1
+        ret[2, 3] = 0
+        return ret
+
+    def initial_conditions(self):
+        """
+        Returns m(0).
```

Comment:
Are these from ToyODEModel ?

Reply:
Not sure I understand? The initial conditions are set in this class's `set_initial_conditions` method.

Reply:
There is no initial_conditions thing in the parent class... We could do this, I suppose. At the moment, it's a bit idiosyncractic because the `initial_conditions` is just a single parameter (we hardcode the other two parameters' initial conditions).

I'm actually thinking it may be better to avoid this ambiguity and set_m0 to make it explicit that we're only referring to one element of the initial conditions. We could then include in the parent class setters / getters for all initial conditions.

Reply:
Just a question :D 
Yeah fair enough if they're local methods then this is fine. Docstring could be slightly longer maybe?

Reply:
I've moved these methods to the parent class since there was a lot of ambiguity over what initial_conditions should / would return. In this class, I've put a getter for m0 only since its an estimated parameter of the model.